services/streammanager/kafka/kafkamanager.go

```diff
@@ -414,11 +456,27 @@ func prepareBatchOfMessages(topic string, batch []map[string]interface{}, timest
 			pkgLogger.Errorf("batch from topic %s is missing the userId attribute", topic)
 			continue
 		}
-
 		marshalledMsg, err := json.Marshal(message)
 		if err != nil {
 			return nil, err
 		}
+		codecs := p.getCodecs()
+		if len(codecs) > 0 {
+			schemaId, _ := data["schemaId"].(string)
+			fmt.Println(schemaId)
+			if schemaId == "" {
+				return nil, fmt.Errorf("schemaId is not available for this event")
```

Comment:
Which "this event"? Can you please include some information in the error that would help people pinpoint the troublesome event?

Reply:
Upfated this error message with the index.